pkg/csv2lp/csvTable.go

```diff
@@ -0,0 +1,464 @@
+package csv2lp
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"sort"
+	"strings"
+	"time"
+	"unsafe"
+)
+
+// column labels using in flux CSV result
+const (
+	labelFieldName   = "_field"
+	labelFieldValue  = "_value"
+	labelTime        = "_time"
+	labelStart       = "_start"
+	labelStop        = "_stop"
+	labelMeasurement = "_measurement"
+)
+
+// types of column with respect to line protocol
+const (
+	linePartIgnored = iota + 1 // ignored in line protocol
+	linePartMeasurement
+	linePartTag
+	linePartField
+	linePartTime
+)
+
+// CsvTableColumn represents processing metadata about a csv column
+type CsvTableColumn struct {
+	// label such as "_start", "_stop", "_time"
+	Label string
+	// "string", "long", "dateTime" ...
+	DataType string
+	// "RFC3339", "2006-01-02"
+	DataFormat string
+	// column's line part (0 means not determined yet), see linePart constants
+	LinePart int
+	// default value to be used for rows where value is an empty string.
+	DefaultValue string
+	// index of this column in the table row, -1 indicates a virtual column
+	Index int
+	// TimeZone of dateTime column, applied when parsing dateTime without timeZone in the format
+	TimeZone *time.Location
+	// parse function, when set,  is used to convert column's string value to interface{}
+	ParseF func(string) (interface{}, error)
+
+	// escaped label for line protocol
+	escapedLabel string
+}
+
+// LineLabel returns escaped name of the column so it can be then used as a tag name or field name in line protocol
+func (c *CsvTableColumn) LineLabel() string {
+	if len(c.escapedLabel) > 0 {
+		return c.escapedLabel
+	}
+	return c.Label
+}
+
+// Value returns the value of the column for the supplied row supplied
+func (c *CsvTableColumn) Value(row []string) string {
+	if c.Index < 0 || c.Index >= len(row) {
+		return c.DefaultValue
+	}
+	val := row[c.Index]
+	if len(val) > 0 {
+		return val
+	}
+	return c.DefaultValue
+}
+
+// CsvColumnError indicates conversion error in a specific column
+type CsvColumnError struct {
+	Column string
+	Err    error
+}
+
+func (e CsvColumnError) Error() string {
+	return fmt.Sprintf("column '%s': %v", e.Column, e.Err)
+}
+
+// CsvTable contains metadata about columns and a state of the CSV processing
+type CsvTable struct {
+	// table columns that extract value from data row
+	columns []CsvTableColumn
```

Comment:
This should become a slice of pointers to the `CsvTableColumn` as you return pointers all over the place.